packages/editor-ui/src/utils/pairedItemUtils.ts

```diff
@@ -165,16 +170,39 @@ function getMapping(paths: { [item: string]: string[][] }): { [item: string]: Se
 	return mapping;
 }
 
-export function getPairedItemsMapping(executionResponse: IExecutionResponse | null): {
+function getItemsCount(runData: IRunData) {
+	let itemsCount = 0;
+	Object.keys(runData).forEach((node) => {
+		runData[node].forEach((taskData) => {
+			const data = taskData.data ?? {};
+			Object.keys(data).forEach((key) => {
+				data[key].forEach((run) => {
+					if (run) {
+						itemsCount += run?.length;
+					}
+				});
+			});
+		});
+	});
```

Comment:
I find this a bit hard to read as it's so dense and really nested. Also, I imagine the `runData` structures can be quite deep and heavy, so it might make sense to micro-optimize a little bit. Using `Object.keys` allocates an array each time, so it might make sense to use `for-in` instead. What do you think about something like this?

```ts
for (node in runData) {
  for (taskData in runData[node]) {
    const data = runData[node][taskData]
    if (!data) continue;

    for (key in data) {
      const runsCount = data[key].reduce((sum, run) => {
         return run ? sum + run.length : sum
      }, 0)
      itemsCount += runsCount
    }
  }
}
```

Reply:
sure thanks for the feedback